src/datarobotx/idp/custom_applications.py

```diff
@@ -68,24 +71,58 @@ def _list_custom_apps(endpoint: str, token: str, name: Optional[str] = None) ->
 
 def _find_existing_custom_app(endpoint: str, token: str, **kwargs: Any) -> str:
     for app in _list_custom_apps(endpoint, token):
-        if all([app[camelize(key)] == kwargs[key] for key in kwargs]):
+        if all([app[camelize(key)] == kwargs[key] for key in kwargs if kwargs[key] is not None]):
```

Comment:
why is this change necessary? what i don't like about it is it opens up the possibility that user explicitly requested None for something but the server has something other than None for that same thing

Reply:
The issue is that env_version_id is being set automatically. If provided or not. And that then fails the test if it wasn't set in the call to get_replace_or_create_custom_app